src/etcd/client.py

```diff
@@ -180,7 +181,58 @@ def uri(protocol, host, port):
         elif password:
             _log.warning('Password provided without username, both are required for authentication')
 
-        self.http = urllib3.PoolManager(num_pools=10, **kw)
+        #############
```

Comment:
Please separate all of this code in a function we will call.

I am also not sure this should be part of the standard "client", to which I'd only add a "proxy" capability.